examples/geostrophic_adjustment.jl

```diff
@@ -0,0 +1,132 @@
+# # Geostrophic adjustment using Oceananigans.HydrostaticFreeSurfaceMode l
+#
+# This example demonstrates how to simulate the one-dimensional geostrophic adjustment of a
+# free surface using `Oceananigans.HydrostaticFreeSurfaceModel`, which solves the hydrostatic
+# Boussinesq equations beneach a small-amplitude free surface, with boundary conditions
+# expanded around ``z = 0`` and free surface dynamics linearized under the assumption
+# ``η / H \ll 1``, where ``η`` is the free surface displacement, and ``H`` is the
+# total depth of the fluid.
+#
+# ## Install dependencies
+#
+# First let's make sure we have all required packages installed.
+
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, JLD2, Plots"
+# ```
+
+# ## A one-dimensional domain
+#
+# We use a one-dimensional domain of geophysical proportions,
+
+using Oceananigans
+using Oceananigans.Utils: kilometers
+
+grid = RegularCartesianGrid(size = (128, 1, 1),
+                            x = (0, 1000kilometers), y = (0, 1000kilometers), z = (-400, 0),
```

Comment:
`y = (0, 1000kilometers)`? this might create confusion.... we have a 1D domain, why do we need to prescribe `y`?

Reply:
Hmm, that's a good point.

Reply:
Once `Flat` works we can omit this kwarg :-/

Reply:
True. But for now wouldn't `y = (0, 1)` be simpler?